### ISSUE TITLE
'role' option ignored when creating a did [v0.4.0]

### ISSUE THREAD

**Issue by ed-curran** [openwallet-foundation/credo-ts#1487]
If i try to create a did using the `IndyVdrModule` module, configured to use `bcovrin:test`, with a call like this:

```js
 await agent.dids.create<IndyVdrDidCreateOptions>({
    method: 'indy',
    secret: {
      seed: TypedArrayEncoder.fromString('<SEED>')
    },
    options: {
      role: 'ENDORSER',
      endorserDid: endorserDid,
      endorserMode: 'internal'
    }
  })
```

The did is created, but the role is unset:

![image](https://github.com/hyperledger/aries-framework-javascript/assets/9150713/aae755e5-ad16-423f-8923-40f6953d7eaf)

From digging it seems indy-vdr does support roles like 'ENDORSER' and 'TRUSTEE'. And the culprit is that `create` (and `parseInput`) in `packages/indy-vdr/src/dids/IndyVdrIndyDidRegistrar.ts` is not passing through the 'role' property. Either the property should be removed or supported I guess.

I would like to use a single initial endorser did (imported) to bootstrap the creation of subsequent dids that can themselves register and anoncred schemas etc.  I suppose thinking about it, i could have the subsequent dids not be endorsers and then use the bootstrap endorser to endorse the transaction performed at schema registration etc somehow? However I'm unsure how to do this with the anoncreds api.

**Comment by swcurran** [comment]
It’s a tricky question. Even if this were fixed, it is unlikely you would achieve what you want.

An Indy Endorser typically does not have the permissions to sign a transaction to create another Endorser. “Typically” as in it is configurable via the Indy Rules, and no Indy instance is ever likely to change the rules so that an Endorser can create another Endorser. Even less likely is that an Endorser can endorse a transaction to create a Trustee DID, since the intention is that a Trustee is more powerful than an Endorser.

An Endorser or a Trustee can be created with a transaction signed with the appropriate set of signatures.  Thus, I’m not sure it is appropriate to not have a role on a Transaction. I just don’t know well enough how the call is made/used.

**Comment by TimoGlastra** [comment]
Thanks for raising this. We should look into the role property being passed through ( @2mau can you take a look? )

> I suppose thinking about it, i could have the subsequent dids not be endorsers and then use the bootstrap endorser to endorse the transaction performed at schema registration etc somehow? However I'm unsure how to do this with the anoncreds api.

@2mau can you add some examples here on how @ed-curran can use the transaction endorsement api in AFJ?

**Comment by ed-curran** [comment]
> An Indy Endorser typically does not have the permissions to sign a transaction to create another Endorser. 

Ah thank you, yeah that's the obvious bit I was missing. In light of that fixing this isn't important for me,  the intuitive behavior would be for the create request  to fail and cite a indy policy violation I think.




> Thanks for raising this. We should look into the role property being passed through ( @2mau can you take a look? )
> 
> > I suppose thinking about it, i could have the subsequent dids not be endorsers and then use the bootstrap endorser to endorse the transaction performed at schema registration etc somehow? However I'm unsure how to do this with the anoncreds api.
> 
> @2mau can you add some examples here on how @ed-curran can use the transaction endorsement api in AFJ?

Thanks, I spotted the `IndyVdrRegisterSchemaInternalOptions` type so I think i get the idea now, using an internal endorser should be pretty easy.

**Comment by 2mau** [comment]
@ed-curran  It seems to me that this is what you are looking for.

https://github.com/hyperledger/aries-framework-javascript/blob/38a0578011896cfcf217713d34f285cd381ad72c/packages/indy-vdr/tests/indy-vdr-did-registrar.e2e.test.ts#L89-L96

You can then get the newly created did like this.
```ts
 const did = didRegistrationResult.didState.did
```

**Comment by 2mau** [comment]
@TimoGlastra I created a PR https://github.com/hyperledger/aries-framework-javascript/pull/1488
The role prope was indeed not included in the nym request

**Comment by ed-curran** [comment]
@djleonskennedy I followed this part about importing a did https://aries.js.org/guides/0.4/tutorials/registering-schema-and-credential-definition#2-importing-a-did-optional

Created a did on http://test.bcovrin.vonx.io/, and used the did and seed (as private key field) when doing `dids.import`

**Comment by djleonskennedy** [comment]
@ed-curran I see thank you, so the endorser's DID is going to be hardcoded in an application, I'm right?

**Comment by ed-curran** [comment]
@djleonskennedy dids can be imported dynamically so doesn't have to be hardcoded, but hardcoded is an easy way to start.

**Comment by TimoGlastra** [comment]
Fixed in #1488
